reyna/src/com/b2msolutions/reyna/Repository.java

```diff
@@ -170,17 +180,21 @@ public void delete(Message message) {
         if (message.getId() == null)
             return;
 
-        SQLiteDatabase db = this.getReadableDatabase();
-        if (!this.doesMessageExist(db, message))
-            return;
+        SQLiteDatabase db = this.getWritableDatabase();
+        try {
+            if (!this.doesMessageExist(db, message))
+                return;
 
-        this.deleteExistingMessage(db, message);
+            this.deleteExistingMessage(db, message);
+        } finally {
+            db.close();
+        }
     }
 
     public void deleteMessagesFrom(long messageId) {
         Logger.v(TAG, "deleteMessagesFrom");
 
-        SQLiteDatabase db = this.getReadableDatabase();
+        SQLiteDatabase db = this.getWritableDatabase();
```

Comment:
I was curious too, according to the docs it returns the writable db unless there's a problem that requires it to open as read only.
http://developer.android.com/reference/android/database/sqlite/SQLiteOpenHelper.html#getReadableDatabase()
@CharlesGarth